pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -0,0 +1,399 @@
+	GetDNSZones(ctx context.Context) ([]DNSZoneResponse, error)
+	GetEncryptionKey(ctx context.Context, keyCRN string) (*EncryptionKeyResponse, error)
+	GetResourceGroups(ctx context.Context) ([]resourcemanagerv2.ResourceGroup, error)
+	GetResourceGroup(ctx context.Context, nameOrID string) (*resourcemanagerv2.ResourceGroup, error)
+	GetSubnet(ctx context.Context, subnetID string) (*vpcv1.Subnet, error)
+	GetVSIProfiles(ctx context.Context) ([]vpcv1.InstanceProfile, error)
+	GetVPC(ctx context.Context, vpcID string) (*vpcv1.VPC, error)
+	GetVPCZonesForRegion(ctx context.Context, region string) ([]string, error)
+	GetZoneIDByName(ctx context.Context, name string) (string, error)
+}
+
+// Client makes calls to the IBM Cloud API.
+type Client struct {
+	managementAPI *resourcemanagerv2.ResourceManagerV2
+	controllerAPI *resourcecontrollerv2.ResourceControllerV2
+	vpcAPI        *vpcv1.VpcV1
+	authenticator *core.IamAuthenticator
+}
+
+// cisServiceID is the Cloud Internet Services' catalog service ID.
+const cisServiceID = "75874a60-cb12-11e7-948e-37ac098eb1b9"
+
+// VPCResourceNotFoundError represents an error for a VPC resoruce that is not found.
+type VPCResourceNotFoundError struct{}
+
+// Error returns the error message for the VPCResourceNotFoundError error type.
+func (e *VPCResourceNotFoundError) Error() string {
+	return "Not Found"
+}
+
+// DNSZoneResponse represents a DNS zone response.
+type DNSZoneResponse struct {
+	// Name is the domain name of the zone.
+	Name string
+
+	// ID is the zone's ID.
+	ID string
+
+	// CISInstanceCRN is the IBM Cloud Resource Name for the CIS instance where
+	// the DNS zone is managed.
+	CISInstanceCRN string
+
+	// CISInstanceName is the display name of the CIS instance where the DNS zone
+	// is managed.
+	CISInstanceName string
+
+	// ResourceGroupID is the resource group ID of the CIS instance.
+	ResourceGroupID string
+}
+
+// EncryptionKeyResponse represents an encryption key response.
+type EncryptionKeyResponse struct{}
+
+// NewClient initializes a client with a session.
+func NewClient() (*Client, error) {
+	apiKey := os.Getenv("IC_API_KEY")
+	authenticator := &core.IamAuthenticator{
+		ApiKey: apiKey,
+	}
+
+	client := &Client{
+		authenticator: authenticator,
+	}
+
+	if err := client.loadSDKServices(); err != nil {
+		return nil, errors.Wrap(err, "failed to load IBM SDK services")
+	}
+
+	return client, nil
+}
+
+func (c *Client) loadSDKServices() error {
+	servicesToLoad := []func() error{
+		c.loadResourceManagementAPI,
+		c.loadResourceControllerAPI,
+		c.loadVPCV1API,
+	}
+
+	// Call all the load functions.
+	for _, fn := range servicesToLoad {
+		if err := fn(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetCISInstance gets a specific Cloud Internet Services instance by its CRN.
+func (c *Client) GetCISInstance(ctx context.Context, crnstr string) (*resourcecontrollerv2.ResourceInstance, error) {
+	_, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
```

Comment:
@BobbyRadford, what's the purpose of creating a child context with a timeout and throwing that context away?  I see this `_, cancel` pattern repeated throughout this file, and I cannot figure out what its purpose is.

Reply:
I think this is just a remnant of early development work that I intended to come back to. The context can and should be passed on to the different API calls made through the various IBM Cloud SDKs.